src/neo/Cryptography/MPT/MPTTrie.cs

```diff
@@ -18,7 +18,7 @@ public partial class MPTTrie<TKey, TValue>
         public MPTTrie(ISnapshot store, UInt256 root)
         {
             this.store = store ?? throw new ArgumentNullException();
-            this.root = root is null ? HashNode.EmptyNode : new HashNode(root);
+            this.root = root is null || root == UInt256.Zero ? HashNode.EmptyNode : new HashNode(root);
```

Comment:
Why is `UInt256.Zero` considered `null`?

Reply:
The state root can be null in neo2, because there is no state in early blocks. I use `UInt256.Zero` to represent in `StateRoot`.
I think we don't need this in neo3.

Reply:
Then we should revert the change.